exporter/datadogexporter/internal/translator/metrics_translator.go

```diff
@@ -329,28 +346,30 @@ func (t *Translator) MapMetrics(md pdata.Metrics) (series []datadog.Metric) {
 				host = fallbackHost
 			}
 		}
-		seenHosts[host] = struct{}{}
+
+		if c, ok := consumer.(HostConsumer); ok {
+			c.ConsumeHost(host)
+		}
```

Comment:
Maybe add a comment explaining that, in the exporter, we make sure that the `Consumer` we use is also a `HostConsumer`, which is why this works (and we don't need to pass a separate `HostConsumer` to `MapMetrics`)?

Reply:
Done!